File: program9.java
//take input from user and return the number of digits in the number

import java.util.*;

class program9              //put the class with main function on the top
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number");
        int iNo = sobj.nextInt();

        Digits nobj = new Digits();

        int iRet = nobj.CountDigits(iNo);

        System.out.println("number is : "+iRet);
    }
}

class Digits
{
    public int CountDigits(int iValue)
    { 
        int iCount = 0;
        int iDigit = 0;

        while(iValue != 0)
        {
            iDigit = iValue % 10;
            iCount++;
            iValue = iValue / 10;
        }
        return iCount;
    }
}

File: program9.cpp
//take input from user(decimal formate) and printf the binary of that number

#include<iostream>
using namespace std;

void DisplayBinary(int iNo)
{
    int iDigit = 0;

    cout<<"Binary conversion of "<<iNo<<" is :"<<"\n";

    while(iNo != 0)
    {
        iDigit = iNo % 2;
        cout<<iDigit;
        iNo = iNo /2;
    }
    cout<<"\n";

}

int main()
{   
    int iValue = 0;

    cout<<"Enter number : "<<"\n";
    cin>>iValue;

    DisplayBinary(iValue);

    return 0;
}